internal/c/libqb.cpp

```diff
@@ -32199,10 +32199,18 @@ int32 addone(int32 x) { return x + 1; } // for testing purposes only
 qbs *func__os() {
     qbs *tqbs;
 #ifdef QB64_WINDOWS
-#    ifdef QB64_32
+#   ifdef QB64_32
+#       ifdef QB64_ARM
+    tqbs = qbs_new_txt("[WINDOWS][32BIT][ARM]");

```

Comment:
You can have the compiler concatenate string constants for you by just placing them one-after-another, so I think we can replace all this logic with something like this:

```c
#ifdef QB64_WINDOWS
#    define QB64_OS_SYSTEM_STR "[WINDOWS]"
#elif defined(QB64_LINUX)
#    define QB64_OS_SYSTEM_STR "[LINUX]"
#elif defined(QB64_MACOSX)
#    define QB64_OS_SYSTEM_STR "[MACOSX][LINUX]"
#else
#    define QB64_OS_SYSTEM_STR ""
#endif

#ifdef QB64_32
#    define QB64_OS_BITS_STR "[32BIT]"
#else
#    define QB64_OS_BITS_STR "[64BIT]"
#endif

#ifdef QB64_ARM
#    define QB64_OS_ARCH_STR "[ARM]"
#else
#    define QB64_OS_ARCH_STR ""
#endif

    // Have the compiler combine all our selections into one string
    tqbs = qbs_new_txt(QB64_OS_SYSTEM_STR QB64_OS_BITS_STR QB64_OS_ARCH_STR);
```

Reply:
Done.